internal/services/machinelearning/machine_learning_registry_resource.go

```diff
@@ -0,0 +1,660 @@
+	RegionDetails                 []RegionDetails                            `tfschema:"region_details"`
+	Tags                          map[string]string                          `tfschema:"tags"`
+	Identity                      []identity.ModelSystemAssignedUserAssigned `tfschema:"identity"`
+	// RegistryPrivateEndpointConnections []RegistryPrivateEndpointConnections       `tfschema:"registry_private_endpoint_connections"`
+}
+
+type MachineLearningRegistryResource struct{}
+
+var _ sdk.ResourceWithUpdate = (*MachineLearningRegistryResource)(nil)
+
+func (m MachineLearningRegistryResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringMatch(regexp.MustCompile(`^[a-zA-z0-9][a-zA-z0-9_-]{2,32}$`), "Registry name must be between 3 and 33 characters long. Its first character has to be alphanumeric, and the rest may contain hyphens and underscores. No whitespace is allowed."),
+		},
+
+		"resource_group_name": commonschema.ResourceGroupName(),
+
+		"identity": commonschema.SystemOrUserAssignedIdentityOptionalForceNew(),
+
+		"location": commonschema.Location(),
+
+		"public_network_access_enabled": {
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+			ForceNew: true,
+			Default:  false,
+		},
+
+		"region_details": {
+			Type:     pluginsdk.TypeList,
+			Required: true,
+			ForceNew: true,
+			MinItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"acr_details": {
+						Type:     pluginsdk.TypeList,
+						Optional: true,
+						Elem: &pluginsdk.Resource{
+							Schema: map[string]*pluginsdk.Schema{
+								"system_created_acr_account": {
+									Type:     pluginsdk.TypeList,
+									Optional: true,
+									Elem: &pluginsdk.Resource{
+										Schema: map[string]*pluginsdk.Schema{
+											"acr_account_name": {
+												Type:     pluginsdk.TypeString,
+												Computed: true,
+											},
+
+											"acr_account_sku": {
+												Type:     pluginsdk.TypeString,
+												Optional: true,
+												ValidateFunc: validation.StringInSlice([]string{
+													"Premium",
+												}, false),
+											},
+
+											"arm_resource_id": {
+												Type:     pluginsdk.TypeString,
+												Computed: true,
+											},
+										},
+									},
+								},
+
+								"user_created_acr_account": {
+									Type:     pluginsdk.TypeList,
+									Optional: true,
+									Elem: &pluginsdk.Resource{
+										Schema: map[string]*pluginsdk.Schema{
+											"arm_resource_id": {
+												Type:         pluginsdk.TypeString,
+												Optional:     true,
+												ValidateFunc: validation.StringIsNotEmpty,
+											},
+										},
+									},
+								},
+							},
+						},
+					},
```

Comment:
There's potential to simplify this a lot and make this more user friendly. In what instances is a container registry created for the user by the service? Do they have to supply the sku type that they want?

Reply:
you are right. users cannot set these fields (`system_created_acr_account` and `user_created_acr_account`) in Portal but the API supports them. should we make the whole `system_created_acr_account` as `Computed`. if no `user_created_acr_account` provided then use the `system_created_acr_account` by default.